benchmarks/benchmarks/bench_io.py

```diff
@@ -81,6 +85,11 @@ def time_comment_loadtxt_csv(self, num_lines):
 
         # inspired by similar benchmark in pandas
         # for read_csv
+
+        # need to rewind StringIO object (unfortunately
+        # confounding timing result somewhat) for every
```

Comment:
Why not do the rewinding at the end of the load, not the start? - that way the first time should cost the same as subsequent times

Reply:
That's an interesting idea--the only reservation I'd have on that is that it may actually be a good thing if the first trial is faster (or as fast as possible) since [timeit docs](https://docs.python.org/3/library/timeit.html#timeit.Timer.repeat) suggest:

>  It’s tempting to calculate mean and standard deviation from the result vector and report these. However, this is not very useful. In a typical case, the lowest value gives a lower bound for how fast your machine can run the given code snippet; higher values in the result vector are typically not caused by variability in Python’s speed, but by other processes interfering with your timing accuracy. So the min() of the result is probably the only number you should be interested in.

I suspect that's also why `asv` does "warm-up" runs to get things cached and so on for fastest possible timings, but I'm happy to change the order if suitable.

Reply:
There are other opinions, see the [discussion](http://perf.readthedocs.io/en/latest/analyze.html#median-and-median-absolute-deviation-vs-mean-and-standard-deviation) in the perf library which prefers the `average +/- std`as a more realistic estimation of how sensitive your code is to machine variations. I think the warm up is for JITs like PyPy or Numba that need time to get going, although there is also [discussion](https://mail.python.org/pipermail/speed/2018-February/000650.html) on the speed.python.org mailing list whether it is valid to discard the warm up time, since it too is part of the time to run a program.